generic/hwinfo.py

```diff
@@ -0,0 +1,135 @@
+        self.run_cmd("hwinfo --isdn")
+        self.run_cmd("hwinfo --joystick")
+        self.run_cmd("hwinfo --keyboard")
+        self.run_cmd("hwinfo --memory")
+        self.run_cmd("hwinfo --modem")
+        self.run_cmd("hwinfo --monitor")
+        self.run_cmd("hwinfo --mouse")
+        self.run_cmd("hwinfo --netcard")
+        self.run_cmd("hwinfo --network")
+        self.run_cmd("hwinfo --partition")
+        self.run_cmd("hwinfo --pci")
+        self.run_cmd("hwinfo --pcmcia")
+        self.run_cmd("hwinfo --pcmcia-ctrl")
+        self.run_cmd("hwinfo --pppoe")
+        self.run_cmd("hwinfo --printer")
+        self.run_cmd("hwinfo --redasd")
+        self.run_cmd("hwinfo --reallyall")
+        self.run_cmd("hwinfo --scanner")
+        self.run_cmd("hwinfo --scsi")
+        self.run_cmd("hwinfo --smp")
+        self.run_cmd("hwinfo --sound")
+        self.run_cmd("hwinfo --storage-ctrl")
+        self.run_cmd("hwinfo --sys")
+        self.run_cmd("hwinfo --tape")
+        self.run_cmd("hwinfo --tv")
+        self.run_cmd("hwinfo --uml")
+        self.run_cmd("hwinfo --usb")
+        self.run_cmd("hwinfo --usb-ctrl")
+        self.run_cmd("hwinfo --vbe")
+        self.run_cmd("hwinfo --wlan")
+        self.run_cmd("hwinfo --xen")
+        self.run_cmd("hwinfo --zip")
+        self.run_cmd("hwinfo --short")
+        self.run_cmd("hwinfo --listmd")
+        disk_name = process.system_output("df -h | egrep '(s|v)d[a-z][1-8]' | "
+                                          "tail -1 | cut -d' ' -f1",
+                                          shell=True).strip("12345")
+        self.run_cmd("hwinfo --disk --only %s" % disk_name)
+        Unique_Id = process.system_output("hwinfo --disk --only /dev/sdb | "
+                                          "grep 'Unique' | head -1 | "
+                                          "cut -d':' -f2", shell=True)
+        self.run_cmd("hwinfo --disk --save-config %s" % Unique_Id)
+        self.run_cmd("hwinfo --disk --show-config %s" % Unique_Id)
+        self.run_cmd("hwinfo --verbose --map")
+        self.run_cmd("hwinfo --all --log FILE")
+        if (not os.path.exists('./FILE')) or (os.stat("FILE").st_size == 0):
+            print "--log option failed"
+            self.is_fail += 1
+        self.run_cmd("hwinfo --dump-db 0")
+        self.run_cmd("hwinfo --dump-db 1")
+        self.run_cmd("hwinfo --version")
+        self.run_cmd("hwinfo --help")
+        self.run_cmd("hwinfo --debug 0 --disk --log=-")
+        self.run_cmd("hwinfo --short --block")
+        self.run_cmd("hwinfo --disk --save-config=all")
+        if "failed" in process.system_output("hwinfo --disk --save-config=all | "
+                                             "grep failed | tail -1", shell=True):
+            self.is_fail += 1
+            print "--save-config option failed"
+        if self.is_fail >= 1:
+            self.fail("%s command(s) failed in hwinfo tool verification" % self.is_fail)
```

Comment:
Instead of telling how many commands failed, it would be good to print which command exactly failed.